core/internal/runsummary/runsummary_test.go

```diff
@@ -27,13 +27,15 @@ func TestExplicitSummary(t *testing.T) {
 		ValueJson: `"abc"`,
 	})
 
-	assert.Equal(t,
-		map[string]any{
-			"x": int64(123),
-			"y": float64(10.5),
-			"z": "abc",
-		},
-		rs.ToMap())
+	encoded, err := rs.Serialize()
+	require.NoError(t, err)
+	assert.JSONEq(t,
```

Comment:
oh that's cool, didn't know about this function! 
super useful, does it handle order correctly?

Reply:
yep! internally it just unmarshals into a map and compares maps. It uses the built-in JSON library, so it doesn't work with Infinity and NaN.

Reply:
we have a few tests in the resume that we should use this one, will make more readable, will update!